web/src/topic_list.ts

```diff
@@ -239,6 +246,32 @@ export class TopicListWidget {
     }
 }
 
+export function filter_topics_by_search_term(topic_names: string[], search_term: string): string[] {
+    if (search_term === "") {
+        return topic_names;
+    }
+
+    const word_separator_regex = /[\s/:_-]/; // Use -, _, :, / as word separators in addition to spaces.
+    const empty_string_topic_display_name = util.get_final_topic_display_name("");
+    return util.filter_by_word_prefix_match(
+        topic_names,
+        search_term,
+        (topic) => (topic === "" ? empty_string_topic_display_name : topic),
+        word_separator_regex,
+    );
+}
```

Comment:
I'm not sure why this gets moved out of `topic_list_data.ts`; it seems like it only interacts with data layer details?

Reply:
Nice catch! Moved it back.